openmeter/billing/adapter/customeroverride.go

```diff
@@ -0,0 +1,239 @@
+		return nil, fmt.Errorf("update customer override: %w", ErrTransactionRequired)
+	}
+
+	update := r.client().BillingCustomerOverride.Update().
+		Where(billingcustomeroverride.CustomerID(input.CustomerID)).
+		SetNillableBillingProfileID(lo.EmptyableToPtr(input.ProfileID)).
+		SetNillableCollectionAlignment(input.Collection.Alignment).
+		SetNillableItemCollectionPeriodSeconds(durationPtrToSecondsPtr(input.Collection.ItemCollectionPeriod)).
+		SetNillableInvoiceAutoAdvance(input.Invoicing.AutoAdvance).
+		SetNillableInvoiceDraftPeriodSeconds(durationPtrToSecondsPtr(input.Invoicing.DraftPeriod)).
+		SetNillableInvoiceDueAfterSeconds(durationPtrToSecondsPtr(input.Invoicing.DueAfter)).
+		SetNillableInvoiceItemResolution(input.Invoicing.ItemResolution).
+		SetNillableInvoiceItemPerSubject(input.Invoicing.ItemPerSubject).
+		SetNillableInvoiceCollectionMethod(input.Payment.CollectionMethod)
+
+	if input.ResetDeletedAt {
+		update = update.ClearDeletedAt()
+	}
+
+	linesAffected, err := update.Save(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if linesAffected == 0 {
+		return nil, billing.NotFoundError{
+			NamespacedID: models.NamespacedID{
+				Namespace: input.Namespace,
+				ID:        input.CustomerID,
+			},
+			Entity: billing.EntityCustomerOverride,
+			Err:    billing.ErrCustomerOverrideNotFound,
+		}
+	}
+
+	// Let's fetch the override with edges
+	dbCustomerOverride, err := r.client().BillingCustomerOverride.Query().
+		Where(billingcustomeroverride.CustomerID(input.CustomerID)).
+		WithBillingProfile(func(bpq *db.BillingProfileQuery) {
+			bpq.WithWorkflowConfig()
+		}).
+		First(ctx)
+	if err != nil {
+		if db.IsNotFound(err) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	return mapCustomerOverrideFromDB(dbCustomerOverride), nil
+}
+
+func (r adapter) GetCustomerOverride(ctx context.Context, input billing.GetCustomerOverrideAdapterInput) (*billing.CustomerOverride, error) {
+	query := r.client().BillingCustomerOverride.Query().
+		Where(billingcustomeroverride.Namespace(input.Namespace)).
+		Where(billingcustomeroverride.CustomerID(input.CustomerID)).
+		WithBillingProfile(func(bpq *db.BillingProfileQuery) {
+			bpq.WithWorkflowConfig()
+		}).
+		WithCustomer()
+
+	if !input.IncludeDeleted {
+		query = query.Where(billingcustomeroverride.DeletedAtIsNil())
+	}
+
+	dbCustomerOverride, err := query.First(ctx)
+	if err != nil {
+		if db.IsNotFound(err) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	if dbCustomerOverride.Edges.Customer == nil {
+		return nil, billing.NotFoundError{
+			NamespacedID: models.NamespacedID{
+				Namespace: input.Namespace,
+				ID:        input.CustomerID,
+			},
+			Entity: billing.EntityCustomer,
+			Err:    billing.ErrCustomerNotFound,
+		}
+	}
+
+	return mapCustomerOverrideFromDB(dbCustomerOverride), nil
+}
+
+func (r adapter) DeleteCustomerOverride(ctx context.Context, input billing.DeleteCustomerOverrideInput) error {
+	err := r.client().BillingCustomerOverride.Update().
```

Comment:
should we check if it updates any?

Reply:
No, as the service already checked that and we are in the same transaction, but changed the code.